src/main/java/io/hops/util/FeaturestoreRestClient.java

```diff
@@ -410,9 +410,21 @@ public static FeaturestoreJdbcConnectorDTO getOnlineFeaturestoreJdbcConnectorRes
       throw new FeaturestoreNotFound("Could not get JDBC Connector for online featurestore:" + featurestore);
     }
     final String responseEntity = response.readEntity(String.class);
-    
-    JSONObject onlineFeaturestoreConnector = new JSONObject(responseEntity);
-    return FeaturestoreHelper.parseJdbcConnectorJson(onlineFeaturestoreConnector);
+
+    JSONArray connectors = new JSONArray(responseEntity);
+    FeaturestoreJdbcConnectorDTO featurestoreJdbcConnectorDTO = null;
+    for (int i = 0; i < connectors.length(); i++) {
+      FeaturestoreJdbcConnectorDTO dto = FeaturestoreHelper.parseJdbcConnectorJson(connectors.getJSONObject(i));
+      if(dto.getName().contains(Constants.ONLINE_JDBC_CONNECTOR_SUFFIX) &&
+              dto.getStorageConnectorType() == FeaturestoreStorageConnectorType.JDBC) {
+        featurestoreJdbcConnectorDTO = dto;
+        break;
```

Comment:
you could just return here and save the `if` clause and the `featurestoreJdbcConnectorDTO` variable I believe